ssh/ssh_tunnel.py

```diff
@@ -0,0 +1,138 @@
+"""
+Module for creating persistent SSH connections for use with synchronous
+commands. Typically, tunnels should be invoked with a context manager to
+ensure proper cleanup. E.G.:
+with contextlib.closing(SSHTunnel(*args, **kwargs)) as tunnel:
+    ssh.write_to_remote('/usr/local/usrpath/testfile.txt', 'test_file.txt')
+    ssh.remote_cmd(['cat', 'test_file.txt'])
+"""
+import logging
+from contextlib import closing
+from subprocess import CalledProcessError, TimeoutExpired, check_call, check_output
+
+logger = logging.getLogger(__name__)
+
+SOCKET_DIR = '$HOME/.cache/ssh'
+SSH_CMD = [
+    '/usr/bin/ssh',
+    '-oConnectTimeout=10',
+    '-oControlMaster=auto',
```

Comment:
Love this control master. Have to point a small issue I have experienced when using persistent connections. If the network connection to the ssh goes down for whatever reason and recovers only after timeout, a subsequent ssh call will hang. The fix for it will be to remove the control file and re-run ssh. Not sure if we will hit that situation here, but will be nice to code around it.